### PR TITLE
unit testing for command json

### PR DESCRIPTION
**Pull request by r-sukanya**

### CODE REVIEW

**Comment by r-sukanya** [comment]
To add command json for second dataset after its been merged in develop

**Review by sebmarste — CHANGES_REQUESTED** [review]

File: server/test/unit/commands.json.test.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                    let historyItemToCompare = historyToCompare[currentHistoryIndex];
+                    expect(historyItem.function).to.be.equal(historyItemToCompare.function);
+                    if (historyToCompare.function === "transformData") {
+                        expect(historyItem.functionParameters).to.deep.equal(historyItemToCompare.functionParameters);
+                    }
+                });
+            }).timeout(5000)
+        });
+    });
+});
```
Comment (sebmarste) [inline server/test/unit/commands.json.test.js]: how do you see if a test was intended to fail or not?

File: server/controllers/nlp.js [hunk 2]
```diff
@@ -54,41 +54,49 @@ exports.handleInput = function (req, res) {
...
-                functionParameters: matchedCommand.parameters,
-                input: input
-            });
-
-            executeFunctions(history, dataset, err => res.status(500).send({
+        module.exports.generateNewHistory(input, dataset, history, (newHistory) => {
+            executeFunctions(newHistory, dataset, err => res.status(500).send({
                 error: err,
-                history: history.splice(-1, 1)
+                history: history
```
[the remark this answers is not in the document]
Comment (r-sukanya) [reply server/controllers/nlp.js]: your logic still holds, as history is everything except new history, which is newHistory

File: server/server.js [hunk 3]
```diff
@@ -35,6 +35,7 @@ app.post('/API/columns', publicController.getColumns);
 app.post('/API/examples', publicController.getExamples);
 app.post('/API/nlp', nlpController.handleInput);
 app.post('/API/classifyTokens', nlpController.classifyTokens);
+app.post('/API/generateNewHistory', nlpController.generateNewHistory);
```
[the remark this answers is not in the document]
Comment (r-sukanya) [reply server/server.js]: correct. i will remove it

File: server/test/unit/commands.json.test.js (hunk 1 above)
Comment (r-sukanya) [reply server/test/unit/commands.json.test.js]: on error, we still get a function : transformData, which is what is being tested for now. I will try to figure out the reject promise soon

Comment (sebmarste) [reply server/test/unit/commands.json.test.js]: ok

File: server/controllers/nlp.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sebmarste) [reply server/controllers/nlp.js]: true, thx

**Review by sebmarste — APPROVED** [review]

**Pull request merged**